hdfs3/core.py

```diff
@@ -672,7 +672,8 @@ def read(self, length=None):
                         buffers.append(p.raw)
                     length -= ret
                 else:
-                    raise IOError('Read file %s Failed:' % self.path, -ret)
+                    msg = ensure_string(_lib.hdfsGetLastError())
+                    raise IOError('Read file %s Failed: %s %s' % self.path, -ret, msg)
```

Comment:
the args should've been wrapped in a tuple. likely to fail.

Reply:
fixed in https://github.com/jettify/hdfs3/pull/8